nistats/_utils/testing.py

```diff
@@ -0,0 +1,140 @@
+import json
+import os
+
+import numpy as np
+import pandas as pd
+from nibabel import Nifti1Image
+
+
+def write_fake_fmri_data(shapes, rk=3, affine=np.eye(4)):
+    mask_file, fmri_files, design_files = 'mask.nii', [], []
+    for i, shape in enumerate(shapes):
+        fmri_files.append('fmri_run%d.nii' % i)
+        data = np.random.randn(*shape)
+        data[1:-1, 1:-1, 1:-1] += 100
+        Nifti1Image(data, affine).to_filename(fmri_files[-1])
+        design_files.append('dmtx_%d.csv' % i)
+        pd.DataFrame(np.random.randn(shape[3], rk),
+                     columns=['', '', '']).to_csv(design_files[-1])
+    Nifti1Image((np.random.rand(*shape[:3]) > .5).astype(np.int8),
+                affine).to_filename(mask_file)
+    return mask_file, fmri_files, design_files
+
+
+def generate_fake_fmri_data(shapes, rk=3, affine=np.eye(4)):
+    fmri_data = []
+    design_matrices = []
+    for i, shape in enumerate(shapes):
+        data = np.random.randn(*shape)
+        data[1:-1, 1:-1, 1:-1] += 100
+        fmri_data.append(Nifti1Image(data, affine))
+        design_matrices.append(pd.DataFrame(np.random.randn(shape[3], rk),
+                                            columns=['a', 'b', 'c']))
+    mask = Nifti1Image((np.random.rand(*shape[:3]) > .5).astype(np.int8),
+                       affine)
+    return mask, fmri_data, design_matrices
+
+
+def _write_fake_bold_img(file_path, shape, rk=3, affine=np.eye(4)):
+    data = np.random.randn(*shape)
+    data[1:-1, 1:-1, 1:-1] += 100
+    Nifti1Image(data, affine).to_filename(file_path)
+    return file_path
+
+
+def _basic_paradigm():
+    conditions = ['c0', 'c0', 'c0', 'c1', 'c1', 'c1', 'c2', 'c2', 'c2']
+    onsets = [30, 70, 100, 10, 30, 90, 30, 40, 60]
+    events = pd.DataFrame({'trial_type': conditions,
+                             'onset': onsets})
+    return events
+
+
+def _basic_confounds(length):
+    columns = ['RotX', 'RotY', 'RotZ', 'X', 'Y', 'Z']
+    data = np.random.rand(length, 6)
+    confounds = pd.DataFrame(data, columns=columns)
+    return confounds
+
+
+def create_fake_bids_dataset(base_dir='', n_sub=10, n_ses=2,
+                             tasks=['localizer', 'main'],
+                             n_runs=[1, 3], with_derivatives=True,
+                             with_confounds=True, no_session=False):
+    """Returns a fake bids dataset directory with dummy files
```

Comment:
prepend an underscore to the function name or write a better docstring ;-)

Reply:
:-D
Understood

Reply:
Should I do both prepend with underscore and improve the docstring?

Reply:
I wouldn't say 'you should' but that's even better if you do both.